lib/service/service.go

```diff
@@ -2206,6 +2207,11 @@ func (process *TeleportProcess) initSSH() error {
 			return trace.Wrap(err)
 		}
 
+		if auditd.IsLoginUIDSet() {
+			log.Warnf("login UID is set, but it shouldn't. Incorrect login UID breaks session ID when using auditd. " +
```

Comment:
Should this be an error? I guess that wouldn't be backward compatible. Will things break if this occurs, or the feature will be disabled?

Reply:
> Should this be an error?

Questionable. One field in the event log will be "incorrect". Each SSH session has a unique session ID but only when the login UID is not set. That makes it easy to trace it. If the login UID is set in the main process, then all sessions share the same session ID, so it's hard to distinguish which event belongs to each session. You can still try to filter by PID or username and at least have partial data.
We cannot also fully disable the feature here. When PAM is enabled, some events are generated outside of Teleport, so disabling anything on our side would make it even more confusing, in my opinion. Then auditd would receive "incorrect" events from PAM and none from Teleport.
The good news is that most people should run Teleport from systemd. Then login UID is correct, and the problem doesn't exist. For those few cases when you run Teleport from bash (probably the most common scenario) as a root we will print that warning.